test/units/modules/network/aci/test_aci_rest.py

```diff
@@ -0,0 +1,299 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2017 Dag Wieers <dag@wieers.com>
+
+# This file is part of Ansible by Red Hat
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+
+import sys
+
+from ansible.compat.tests import unittest
+from ansible.modules.network.aci.aci_rest import aci_response
+
+from nose.plugins.skip import SkipTest
+
+from lxml import etree
+if sys.version_info >= (2, 7):
+    from xmljson import cobra
+
+
+class AciRest(unittest.TestCase):
+
+    def test_invalid_aci_login(self):
+        self.maxDiff = None
+
+        expected_result = {
+            u'error_code': u'401',
+            u'error_text': u'Username or password is incorrect - FAILED local authentication',
+            u'imdata': [{
+                u'error': {
+                    u'attributes': {
+                        u'code': u'401',
+                        u'text': u'Username or password is incorrect - FAILED local authentication',
+                    },
+                },
+            }],
+            u'totalCount': '1',
+        }
+
+        json_response = '{"totalCount":"1","imdata":[{"error":{"attributes":{"code":"401","text":"Username or password is incorrect - FAILED local authentication"}}}]}'  # NOQA
+        json_result = aci_response(json_response, 'json')
+        self.assertEqual(expected_result, json_result)
+
+        # Python 2.7+ is needed for xmljson
+        if sys.version_info < (2, 7):
+            return
+
+        xml_response = '''<?xml version="1.0" encoding="UTF-8"?><imdata totalCount="1">
```

Comment:
fixture with load_json. If it's not valid json then load_json just gives you back a plain string